markdown/docs/addon-author-guide.md

```diff
@@ -0,0 +1,321 @@
+# Addon Author Guide
+
+If you're the author of an Ember addon, you might need to make some
+changes to ensure that your addon runs correctly in the FastBoot
+environment.
+
+Remember that when updating your addons for FastBoot compatibility,
```

Comment:
@arjansingh I was going to extract some of the material here to an "App Upgrade Guide," which shares a lot of material, but there is some addon-specific stuff I don't want to overwhelm people.